File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        StringBuilder leftSide = new StringBuilder();
        
        // 각 음식의 절반을 왼쪽에서 먹을 수 있도록 배치
        for (int i = 1; i < food.length; i++) {
            int count = food[i] / 2;  // 각 음식의 절반을 가져옴
            for (int j = 0; j < count; j++) {
                leftSide.append(i);  // 왼쪽에서 먹는 음식을 추가
            }
        }
        
        // 왼쪽 음식 + 물(0) + 오른쪽 음식을 만들어 반환
        String rightSide = leftSide.reverse().toString();  // 왼쪽 음식을 뒤집어서 오른쪽에 배치
        return leftSide.reverse().toString() + "0" + rightSide;
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.kt
class Solution {
    fun solution(food: IntArray): String {
        val left = StringBuilder()

        for (i in 1 until food.size) {
            val count = food[i] / 2  // 양쪽에 똑같이 배치할 수 있는 개수
            repeat(count) {
                left.append(i)
            }
        }

        val right = left.toString().reversed()

        return left.toString() + "0" + right
    }
}
